backends/mlu/kernels/kldiv_loss_kernel.cc

```diff
@@ -0,0 +1,175 @@
+  sub_out.Resize(x.dims());
+  dev_ctx.template Alloc<T>(&sub_out);
+  MLUOpTensorKernel<T>(dev_ctx, log_label, x, -1, CNNL_OP_TENSOR_SUB, &sub_out);
+  // 3. optensor-- mul(label,sub_out) -->out_tmp
+  MLUOpTensorKernel<T>(
+      dev_ctx, label, sub_out, -1, CNNL_OP_TENSOR_MUL, &out_tmp);
+
+  MLUCnnlTensorDesc out_tmp_desc(out_tmp);
+  MLUCnnlTensorDesc condition_desc(condiction_out);
+  MLUCnnl::Select(dev_ctx,
+                  condition_desc.get(),
+                  GetBasePtr(&condiction_out),
+                  out_tmp_desc.get(),
+                  GetBasePtr(&out_tmp),
+                  tensor_zeros_desc.get(),
+                  GetBasePtr(&tensor_zeros),
+                  out_tmp_desc.get(),
+                  GetBasePtr(&out_tmp));
+
+  std::vector<int64_t> axes = {-1};
+  // 4.reduction
+  if ("none" == reduction) {
+    *out = out_tmp;
+  } else if ("batchmean" == reduction) {
+    const int batch = x.dims()[0];
+    float alpha = 1.0 / batch;
+    float beta = 0;
+    MLUCnnlTensorDesc out_desc(*out);
+    MLUReduceOp<T>(dev_ctx, out_tmp, axes, false, true, "reduce_sum", out);
+    MLUCnnl::Transform(dev_ctx,
+                       &alpha,
+                       &beta,
+                       out_desc.get(),
+                       GetBasePtr(out),
+                       out_desc.get(),
+                       GetBasePtr(out));
+  } else if ("mean" == reduction) {
+    MLUReduceOp<T>(dev_ctx, out_tmp, axes, false, true, "reduce_mean", out);
+  } else if ("sum" == reduction) {
+    MLUReduceOp<T>(dev_ctx, out_tmp, axes, false, true, "reduce_sum", out);
+  }
+}
+
+template <typename T, typename Context>
+void KLDivLossGradKernel(const Context& dev_ctx,
+                         const phi::DenseTensor& x,
+                         const phi::DenseTensor& label,
+                         const phi::DenseTensor& d_out,
+                         const std::string& reduction,
+                         phi::DenseTensor* d_x) {
+  dev_ctx.template Alloc<T>(d_x);
+  // dx = -1 * label * d_out
+  //  Relu(label) make label >=0
+  phi::DenseTensor label_clip;
+  label_clip.Resize(label.dims());
+  dev_ctx.template Alloc<T>(&label_clip);
+  MLUCnnlActivationDesc act_desc(CNNL_ACTIVATION_RELU, 1.0);
+  MLUCnnlTensorDesc label_desc(label);
+  MLUCnnlTensorDesc label_clip_desc(label_clip);
+
+  MLUCnnl::Active(dev_ctx,
+                  act_desc.get(),
+                  label_desc.get(),
+                  GetBasePtr(&label),
+                  label_clip_desc.get(),
+                  GetBasePtr(&label_clip));
+
+  // label * d_out
+  phi::DenseTensor out_tmp;
+  out_tmp.Resize(x.dims());
+  dev_ctx.template Alloc<T>(&out_tmp);
+  MLUOpTensorKernel<T>(
+      dev_ctx, label_clip, d_out, -1, CNNL_OP_TENSOR_MUL, &out_tmp);
+
+  MLUCnnlTensorDesc out_tmp_desc(out_tmp);
+
+  float alpha = -1.0;
+  float beta = 0;
+  if ("mean" == reduction) {
+    alpha = static_cast<float>(-1.0 / d_x->numel());
+  } else if ("batchmean" == reduction) {
```

Comment:
Is it necessary to take care of other cases shown in the fwd-kernel? If not, please leave a comment.